rpc/src/v1/impls/eth_pubsub.rs

```diff
@@ -141,19 +141,20 @@ impl<C> ChainNotificationHandler<C> {
 		}
 	}
 
-	fn notify_logs<F, T>(&self, enacted: &[H256], logs: F) where
-		F: Fn(EthFilter) -> T,
+	fn notify_logs<F, T, Ex>(&self, enacted: &[(H256, Ex)], logs: F) where
```

Comment:
It looks like we cannot change `enacted: &[(H256, Ex)]` to an generic `Iterator`. We need to iterate over `enacted` many times, but most `Iterator` are not clonable. This do mean that in `LightChainNotify`, there's one additional `collect::Vec<_>`, but I think this tradeoff is okay.